admin/templates/configuration/etc/xrdssi.cnf

```diff
@@ -28,9 +28,8 @@ location = {{QSERV_DATA_DIR}}/mysql
 [scheduler]
 
 # Thread pool size
-# thread_pool_size = 20
 #thread_pool_size = 12
-thread_pool_size = 100
+thread_pool_size = 21
```

Comment:
I changed it to 20, but finding the best value requires experimenting. More threads could be better, but contention can become issues pretty easily.

Reply:
Multi-parametric optimization is a hard problem due to various indirect dependencies between parameters - changing a number of threads may require changing buffer sizes, a size of the MySQL connection pool, etc. Besides, a lot of things may depend on a hardware where Qserv is installed, a size of Qserv itself, as well as on the sizes and complexity of the catalogs. So, what's set for one unique combination of conditions may not work for others.

It would be nice to have just a few parameters which would have a predicted behavior ... or let Qserv to dynamically tune these parameters depending on the observed performance, latencies, resource utilization. This is just a "wish list".

Reply:
I think the problem is really complicated and may need machine learning to come up with an optimal solution. At least having them in the config files makes it easier to experiment and find decent answers.